src/Form/formatters.js

```diff
@@ -110,10 +110,145 @@ const getCountryName = (el, options = {}) => {
     return localisedCountryName
 }
 
+/**
+ * Try to get a map of localised country names to code, or falls back to the English map
+ * @param {HTMLInputElement | HTMLSelectElement} el
+ */
+const getLocalisedCountryNamesToCodes = (el) => {
+    if (typeof Intl.DisplayNames !== 'function') return COUNTRY_NAMES_TO_CODES
+
+    // Try to infer the field language or fallback to en
+    const elLocale = inferElementLocale(el)
+
+    return Object.fromEntries(
+        Object.entries(COUNTRY_CODES_TO_NAMES)
+            .map(([code]) => [getCountryDisplayName(elLocale, code), code])
+    )
+}
+
+/**
+ * Try to infer a country code from an element we identified as identities.addressCountryCode
+ * @param {HTMLInputElement | HTMLSelectElement} el
+ * @return {string}
+ */
+const inferCountryCodeFromElement = (el) => {
+    if (COUNTRY_CODES_TO_NAMES[el.value]) return el.value
+    if (COUNTRY_NAMES_TO_CODES[el.value]) return COUNTRY_NAMES_TO_CODES[el.value]
+
+    const localisedCountryNamesToCodes = getLocalisedCountryNamesToCodes(el)
+    if (localisedCountryNamesToCodes[el.value]) return localisedCountryNamesToCodes[el.value]
+
+    if (el instanceof HTMLSelectElement) {
+        const selectedText = el.selectedOptions[0]?.text
+        if (COUNTRY_CODES_TO_NAMES[selectedText]) return selectedText
+        if (COUNTRY_NAMES_TO_CODES[selectedText]) return localisedCountryNamesToCodes[selectedText]
+        if (localisedCountryNamesToCodes[selectedText]) return localisedCountryNamesToCodes[selectedText]
+    }
+    return ''
+}
+
+/**
+ * @param {InternalDataStorageObject} credentials
+ * @return {boolean}
+ */
+const shouldStoreCredentials = ({credentials}) =>
+    Boolean(credentials.password)
+
+/**
+ * @param {InternalDataStorageObject} credentials
+ * @return {boolean}
+ */
+const shouldStoreIdentities = ({identities}) =>
+    Boolean(
+        (identities.firstName || identities.fullName) &&
+        identities.addressStreet &&
+        identities.addressCity
+    )
+
+/**
+ * @param {InternalDataStorageObject} credentials
+ * @return {boolean}
+ */
+const shouldStoreCreditCards = ({creditCards}) =>
+    Boolean(creditCards.cardNumber && (
```

Comment:
I think these chained conditions are at the point where they could benefit for being expanded out, something like 

```ts
const shouldStoreCreditCards = ({creditCards}) => {
    if (!creditCards.cardNumber) return false;
    if (creditCards.cardSecurityCode) return true;
    if (creditCards.cardName) {
        if (creditCards.expiration) return true;
        return creditCards.expirationYear && creditCards.expirationMonth;
    }
    return false;
}
```

should make it easier for others to visually parse/understand the logic here.

I don't want to block the PR, but let's make a follow-up on this.

Reply:
I've added tests and implemented your suggestion. Looks much better now!